File: Assets/Typescript~/src/Server/Services/Global/Chat/Commands/TpAllCommand.ts
import { Dependency } from "@easy-games/flamework-core";
import { ChatCommand } from "Imports/Core/Shared/Commands/ChatCommand";
import { Player } from "Shared/Player/Player";
import { PlayerService } from "../../Player/PlayerService";

export class TpAllCommand extends ChatCommand {
	constructor() {
		super("tpAll");
	}

	public Execute(player: Player, args: string[]): void {
		const pos = player.Character?.gameObject.transform.position;
		if (!pos) return;

		for (const p of Dependency<PlayerService>().GetPlayers()) {
			if (p !== player) {
				if (p.Character) {
					const humanoid = p.Character.gameObject.GetComponent<EntityDriver>();
					humanoid.Teleport(pos);
				}
			}
		}
	}
}

File: Assets/Bundles/Server/Resources/TS/Services/Global/Chat/Commands/TpAllCommand.lua
-- Compiled with unity-ts v2.1.0-75
local Flamework = require("Shared/rbxts_include/node_modules/@easy-games/flamework-core/out/init").Flamework
local ChatCommand = require("Server/TS/Services/Global/Chat/Commands/ChatCommand").ChatCommand
local TpAllCommand
do
	local super = ChatCommand
	TpAllCommand = setmetatable({}, {
		__tostring = function()
			return "TpAllCommand"
		end,
		__index = super,
	})
	TpAllCommand.__index = TpAllCommand
	function TpAllCommand.new(...)
		local self = setmetatable({}, TpAllCommand)
		return self:constructor(...) or self
	end
	function TpAllCommand:constructor()
		super.constructor(self, "tpAll")
	end
	function TpAllCommand:Execute(player, args)
		local _pos = player.Character
		if _pos ~= nil then
			_pos = _pos.gameObject.transform.position
		end
		local pos = _pos
		if not pos then
			return nil
		end
		for _, p in (Flamework.resolveDependency("Bundles/Server/Services/Global/Player/PlayerService@PlayerService")):GetPlayers() do
			if p ~= player then
				if p.Character then
					local humanoid = p.Character.gameObject:GetComponent("EntityDriver")
					humanoid:Teleport(pos)
				end
			end
		end
	end
end
return {
	TpAllCommand = TpAllCommand,
}
-- ----------------------------------
-- ----------------------------------
